packages/block-editor/src/components/url-popover/image-url-input-ui.js

```diff
@@ -0,0 +1,312 @@
+				title: __( 'Attachment Page' ),
+				url: mediaType === 'image' ? mediaLink : undefined,
+				icon: <SVG viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><Path d="M0 0h24v24H0V0z" fill="none" /><Path d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z" /></SVG>,
+			},
+		];
+	};
+
+	const onSetHref = ( value ) => {
+		const linkDestinations = getLinkDestinations();
+		let linkDestinationInput;
+		if ( ! value ) {
+			linkDestinationInput = LINK_DESTINATION_NONE;
+		} else {
+			linkDestinationInput = (
+				find( linkDestinations, ( destination ) => {
+					return destination.url === value;
+				} ) ||
+				{ linkDestination: LINK_DESTINATION_CUSTOM }
+			).linkDestination;
+		}
+		onChangeUrl( {
+			linkDestination: linkDestinationInput,
+			href: value,
+		} );
+	};
+
+	const onSetNewTab = ( value ) => {
+		const updatedLinkTarget = getUpdatedLinkTargetSettings( value );
+		onChangeUrl( updatedLinkTarget );
+	};
+
+	const onSetLinkRel = ( value ) => {
+		onChangeUrl( { rel: value } );
+	};
+
+	const onSetLinkClass = ( value ) => {
+		onChangeUrl( { linkClass: value } );
+	};
+
+	const advancedOptions = (
+		<>
+			<ToggleControl
+				label={ __( 'Open in New Tab' ) }
+				onChange={ onSetNewTab }
+				checked={ linkTarget === '_blank' } />
+			<TextControl
+				label={ __( 'Link Rel' ) }
+				value={ removeNewTabRel( rel ) || '' }
+				onChange={ onSetLinkRel }
+				onKeyPress={ stopPropagation }
+				onKeyDown={ stopPropagationRelevantKeys }
+			/>
+			<TextControl
+				label={ __( 'Link CSS Class' ) }
+				value={ linkClass || '' }
+				onKeyPress={ stopPropagation }
+				onKeyDown={ stopPropagationRelevantKeys }
+				onChange={ onSetLinkClass }
+			/>
+		</>
+	);
+
+	const linkEditorValue = urlInput !== null ? urlInput : url;
+
+	const urlLabel = ( find( getLinkDestinations(), [ 'linkDestination', linkDestination ] ) || {} ).title;
+
+	return (
+		<>
+			<IconButton
+				icon="admin-links"
+				className="components-toolbar__control"
+				label={ url ? __( 'Edit link' ) : __( 'Insert link' ) }
+				aria-expanded={ isOpen }
+				onClick={ openLinkUI }
+			/>
+			{ isOpen && (
+				<URLPopover
+					onFocusOutside={ onFocusOutside() }
+					onClose={ closeLinkUI }
+					renderSettings={ () => advancedOptions }
+					additionalControls={ ! linkEditorValue && (
+						<NavigableMenu>
+							{
+								map( getLinkDestinations(), ( link ) => (
+									<MenuItem
+										key={ link.linkDestination }
+										icon={ link.icon }
+										onClick={ () => {
+											setUrlInput( null );
+											onSetHref( link.url );
+											stopEditLink();
+										} }
+									>
+										{ link.title }
+									</MenuItem>
+								) )
+							}
+						</NavigableMenu>
+					) }
+				>
+					{ ( ! url || isEditingLink ) && (
+						<URLPopover.LinkEditor
+							className="editor-format-toolbar__link-container-content block-editor-format-toolbar__link-container-content"
```

Comment:
This is not a valid class name for this component:

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming

It violates the guideline to not inherit styles from other compoennts:

>A component's class name should never be used outside its own folder (with rare exceptions such as _z-index.scss). If you need to inherit styles of another component in your own components, you should render an instance of that other component. At worst, you should duplicate the styles within your own component's stylesheet. This is intended to improve maintainability by isolating shared components as a reusable interface, reducing the surface area of similar UI elements by adapting a limited set of common components to support a varied set of use-cases.

It also introduces additional legacy compatibility class names intended to have been removed as part of #19050 (though I also note that this is problematic in the `format-library` from which this code was presumably copied).

Reply:
I'll make a PR ASAP :)